src/ncp/ncp_base_mtd.cpp

```diff
@@ -679,7 +679,7 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_THREAD_ALLOW_LOCAL_NE
     SuccessOrExit(error = mDecoder.ReadBool(value));
 
     // Register any net data changes on transition from `true` to `false`.
-    shouldRegisterWithLeader = (mAllowLocalNetworkDataChange == true) && (value == false);
+    shouldRegisterWithLeader = (mAllowLocalNetworkDataChange) && (!value);
```

Comment:
```suggestion
    shouldRegisterWithLeader = mAllowLocalNetworkDataChange && !value;
```
nit: remove redundant () ?